chain/network/src/types.rs

```diff
@@ -186,6 +210,49 @@ impl Handshake {
     }
 }
 
+impl From<HandshakeV1> for Handshake {
+    fn from(handshake_old: HandshakeV1) -> Self {
+        Self {
+            version: handshake_old.version,
+            oldest_supported_version: handshake_old.version - 1,
```

Comment:
why?

Reply:
This is not relevant, it is assuming that versions before 30 were compatible with the previous versions, but this is just a default value and doesn't change anything.

Reply:
It seems that a peer can crash you by sending a handshake with version 0?

Reply:
good catch, will use `oldest_supported_version: handshake_old.version` since it doesn't really change any behavior.